static/src/javascripts/projects/common/modules/identity/modules/fetchFormFields.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+import fastdom from 'lib/fastdom-promise';
+
+export const ERR_IDENTITY_HTML_PREF_NOT_FOUND = `Can't find HTML preference`;
+
+export const getCsrfTokenFromElement = (
+    originalEl: ?HTMLElement
+): Promise<any> =>
+    fastdom
+        .read(() => {
+            if (!originalEl) return Promise.reject();
+            const closestFormEl: ?Element = originalEl.closest('form');
+            if (closestFormEl) {
+                return closestFormEl.querySelector('*[name=csrfToken]');
```

Comment:
Does it need the asterix?

Reply:
![](http://25.media.tumblr.com/tumblr_mckz1uL9nK1ryr815o1_500.gif)